### ISSUE TITLE
Multiple issues with JAWS

### ISSUE THREAD

**Issue by lmulhall-phet** [phetsims/resistance-in-a-wire#132]
**Test device:**

Hanson

**Operating System:**

Windows 10

**Browser:**

Firefox ESR 52.6.0

**Problem description:**

NVDA, JAWS, and VoiceOver aren't behaving similarly. NVDA and JAWS are, I think, not behaving how they're supposed to. JAWS is much worse than NVDA in this regard. I hope this issue can be an overview/outline of the issues we find with JAWS and NVDA. I already made an issue for a specific odd behavior I found in JAWS (phetsims/resistance-in-a-wire/issues/131). I think we should try to keep track of all of these odd behaviors in this issue. This is for phetsims/QA/issues/92.

Please feel free to add any issues you encounter with any of the screen readers.

VoiceOver > NVDA > JAWS.

1. JAWS doesn't read the description of the simulation. "This is an interactive simulation..." NVDA read the description once. VoiceOver always reads the description.
2. JAWS does read the resistivity, length, or area as they are increased or decreased. However, JAWS will not read out the relationship between the variables. For instance, it will not say "As letter L increases, letter R increases." This is not the case with NVDA or VoiceOver.
3. JAWS will say "tab" when tab is pressed instead of saying what I've tabbed to, e.g. the resistivity slider. This is not the case with NVDA or VoiceOver.
4. JAWS has multiple readouts per increment. See phetsims/resistance-in-a-wire/issues/131.

**Troubleshooting information (do not edit):**

Name: ‪Resistance in a Wire‬
URL: https://www.colorado.edu/physics/phet/dev/html/resistance-in-a-wire/1.5.0-dev.1/phet/resistance-in-a-wire_en_phet.html
Version: 1.5.0-dev.1 2018-02-09 20:00:00 UTC
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:52.0) Gecko/20100101 Firefox/52.0
Language: en-US
Window: 1280x611
Pixel Ratio: 2.5/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0
Vendor: Mozilla (Mozilla)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"928741cf","branch":"master"},"axon":{"sha":"e1350857","branch":"master"},"brand":{"sha":"53cf227b","branch":"master"},"chipper":{"sha":"10125b6e","branch":"master"},"dot":{"sha":"8d471f40","branch":"master"},"joist":{"sha":"5567ef56","branch":"master"},"kite":{"sha":"b6071478","branch":"master"},"phet-core":{"sha":"f35ff65e","branch":"master"},"phet-io":{"sha":"59e4fea1","branch":"master"},"phet-io-website":{"sha":"766da99e","branch":"master"},"phet-io-wrapper-classroom-activity":{"sha":"c84e3046","branch":"master"},"phet-io-wrapper-lab-book":{"sha":"ebf7c7dc","branch":"master"},"phet-io-wrapper-sonification":{"sha":"c276b0fa","branch":"master"},"phet-io-wrapper-visualizing-engagement":{"sha":"92165486","branch":"master"},"phet-io-wrappers":{"sha":"d79148b8","branch":"master"},"phetcommon":{"sha":"d33be800","branch":"master"},"query-string-machine":{"sha":"485e174e","branch":"master"},"resistance-in-a-wire":{"sha":"5031adb2","branch":"master"},"scenery":{"sha":"c1725c3e","branch":"master"},"scenery-phet":{"sha":"666a80bb","branch":"master"},"sherpa":{"sha":"7a6906fd","branch":"master"},"sun":{"sha":"b854de6b","branch":"master"},"tandem":{"sha":"63c71e1a","branch":"master"}}

**Comment by lmulhall-phet** [comment]
JAWS is behaving differently (more or less correctly) today. Not sure if settings have been altered.

**Comment by lmulhall-phet** [comment]
Did notice that in the scene summary, JAWS will read out "rho is 0.50 bullet ..." Wondering why "bullet" is in there.

**Comment by lmulhall-phet** [comment]
Also noticed that using keyboard navigation and JAWS leads to some performance issues if the sliders aren't increased or decreased in discrete steps. In other words, if you press and hold an arrow key to increase or decrease length, for instance, there will be some lag between input and the sim moving the slider up or down.

**Comment by lmulhall-phet** [comment]
Update: NVDA is behaving correctly. I did some testing yesterday and today. I think the NVDA issues were related to settings. JAWS, on the other hand, is not behaving correctly. Here are the issues I'm seeing with JAWS:
- Not reading sim description
- Not reading "as x grows, y shrinks"
- Reading "tab" when tab is pressed
- Reading values twice, i.e. "11.00 cm, to increase or decrease use the arrow keys, 11.00 cm" or "11.00 cm, 11.00 cm, to increase or decrease use the arrow keys"
- Reading "to increase or decrease use the arrow keys"
- Still seeing some performance issues when adjusting the sliders while using JAWS

**Comment by lmulhall-phet** [comment]
Did some VoiceOver testing this morning, and I'm wondering why VoiceOver isn't reading the sim description.

**Comment by jessegreenberg** [comment]
> Did some VoiceOver testing this morning, and I'm wondering why VoiceOver isn't reading the sim description.

Regarding these kinds of issues,  the screen reader might not read the entire sim every page load. @lmulhall-phet are you still able to read the sim description with the reading keys?

**Comment by lmulhall-phet** [comment]
@jessegreenberg Yes, sorry. My fault. Difficult to transition between the screen readers. VoiceOver looks good.

**Comment by jessegreenberg** [comment]
@lmulhall-phet the same question applies to JAWS, are you able to read the sim description with reading keys, even if it doesn't read through  on load?

**Comment by lmulhall-phet** [comment]
@jessegreenberg using default settings, I'm unable to get JAWS to read the sim description using insert+down arrow, insert+B, caps lock+a, or caps lock+down arrow.

**Comment by jessegreenberg** [comment]
Sorry @lmulhall-phet I am not able to reproduce this. I just tried with JAWS and Firefox, I am able to read through  the scene summary, hear the correct alerts about the equation, and so on. I will try on hanson tomorrow.

**Comment by lmulhall-phet** [comment]
@jessegreenberg then if I had to guess, I’d say it’s probably just ineptitude on my part. I probably should have spent more time getting familiar with the JAWS keystrokes and screen readers in general before logging the issue. Sorry about that.

**Comment by jessegreenberg** [comment]
No worries at all @lmulhall-phet, @phet-steele and I were able to reproduce the problem today.

@phet-steele and I played with the sim. At first the problem was there. We restarted FF and the problem went away. Then he was able to make the problem resurface after doing both of these

- Restarting computer
- Navigating the sim before turning JAWS on.

We aren't sure which one is causing the problem yet.

**Comment by jessegreenberg** [comment]
I just did the above steps on my machine and encountered the same problem. Restarting FF made the problem go away. Going to try to reproduce with another non-PhET webpage.

**Comment by jessegreenberg** [comment]
I did the above steps with reddit.com, and encountered the same problem. This is not a PhET problem.

**Comment by jessegreenberg** [comment]
I tried insert+z to toggle virtual cursor on and off, had no impact. We should submit a bug report to FF.

**Comment by jessegreenberg** [comment]
I submitted a bug report to Firefox about this issue in the above a11y-research issue. Thanks @lmulhall-phet and @phet-steele for reporting this problem and looking into it. It sounds like the best way to prevent this is to

- Open JAWS, wait for screen reader to fully boot
- Boot up Firefox
- Navigate to webpage

Any other order may or may not work with JAWS.

Closing.
